libraries/AP_HAL/utility/Socket.cpp

```diff
@@ -106,11 +115,30 @@ ssize_t SocketAPM::send(void *buf, size_t size)
 /*
   send some data
  */
-ssize_t SocketAPM::sendto(void *buf, size_t size, const char *address, uint16_t port)
+ssize_t SocketAPM::sendto(const void *buf, size_t size, const char *address, uint16_t port, uint32_t timeout)
 {
-    struct sockaddr_in sockaddr;
-    make_sockaddr(address, port, sockaddr);
-    return ::sendto(fd, buf, size, 0, (struct sockaddr *)&sockaddr, sizeof(sockaddr));
+    struct pollfd pollfd = {
+        .fd = fd,
+        .events = POLLOUT,
+    };
+
+    int ret = poll(&pollfd, 1, timeout);
+    
+    if (ret > 0) {
+        if (pollfd.revents & POLLOUT) {
+            struct sockaddr_in sockaddr;
+            make_sockaddr(address, port, sockaddr);
+            socklen_t socklen = sizeof(sockaddr);
+
+            return ::sendto(fd, buf, size, 0, (struct sockaddr *)&sockaddr, socklen);
+        }
+    } else if (ret == -1) {
+        perror("poll");
```

Comment:
we can't have error messages that may repeat. If a socket buffer continually fills up this perror() will fire continuously, spamming the console